File: TreasurersApp/ClientApp/src/app/models/ContributorActionResult.ts
import { Contributor } from './Contributor';
import { ContributorService } from '../maintenance/contributor-maintenance/contributor.service';

export class ContributorActionResult {
  success = false;
  statusMessages: string[] = [];
  contributor: Contributor = null;

  constructor(private contributorService: ContributorService) {
    this.success = false;
    this.contributor = this.contributorService.newContributor();
    this.statusMessages = [];
  }
}

File: TreasurersApp/Models/ContributorActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class ContributorActionResult : WebResult<Contributor>
    {
        public ContributorActionResult(bool success, List<string> statusMessages, Contributor contributor)
            :base(success, statusMessages, contributor)
        {
        }
    }
}
